arangod/IResearch/IResearchLinkHelper.cpp

```diff
@@ -55,6 +55,39 @@
 
 namespace {
 
+bool isIgnoredHiddenEnterpriseCollection(std::string const& cName) {
+  /*
+   * Note: As IResearchView.cpp L204 says:
+   * -> "create links on a best-effort basis, link creation failure does not cause view creation failure"
+   *
+   * Workaround: If we detect a collection which should not be created in the SingleServer case,
+   * let the link validation itself return a success.
+   *
+   * Nevertheless, the user will be notified that there has been an edge case. This should be fine.
+   * Another attempt could be to rewrite the links itself, but more code changes will then be
+   * necessary.
+   */
+#ifdef USE_ENTERPRISE
+  if (arangodb::ServerState::instance()->isSingleServer()) {
+    if (cName[0] == '_') {
+      if (strncmp(cName.c_str(), arangodb::StaticStrings::FullLocalPrefix.c_str(), 7) == 0 ||
+          strncmp(cName.c_str(), arangodb::StaticStrings::FullFromPrefix.c_str(), 6) == 0 ||
+          strncmp(cName.c_str(), arangodb::StaticStrings::FullToPrefix.c_str(), 4) == 0) {
+        LOG_TOPIC("d921b", INFO, arangodb::Logger::VIEWS)
+            << "Ignoring view '" << cName
+            << "'. Will only be initially created via SmartGraphs of a full "
+               "dump of a cluster."
+               "This view is supposed to be not restored in case you dump from "
+               "and cluster and "
+               "then restore into a single-server instance.";
```

Comment:
Please, reformulate, e.g.,
"'. This view is not supposed to be restored in case you dump from "
               "a cluster and "
               "then restore into a single-server instance.";
(I don't understand the first sentence; and -> a.)

Reply:
Improved comment.